akka-http-core/src/main/java/akka/http/shaded/com/twitter/hpack/Decoder.java

```diff
@@ -464,44 +464,47 @@ private void readName(int index) throws IOException {
   private void indexHeader(int index, HeaderListener headerListener) throws IOException {
     if (index <= StaticTable.length) {
       HeaderField headerField = StaticTable.getEntry(index);
-      addHeader(headerListener, headerField.name, headerField.value, false);
+      Object parsed = addHeader(headerListener, headerField.name, headerField.value, headerField.parsedValue, false);
+      headerField.parsedValue = parsed;
     } else if (index - StaticTable.length <= dynamicTable.length()) {
       HeaderField headerField = dynamicTable.getEntry(index - StaticTable.length);
-      addHeader(headerListener, headerField.name, headerField.value, false);
+      Object parsed = addHeader(headerListener, headerField.name, headerField.value, headerField.parsedValue, false);
+      headerField.parsedValue = parsed;
     } else {
       throw ILLEGAL_INDEX_VALUE;
     }
   }
 
-  private void insertHeader(HeaderListener headerListener, byte[] name, byte[] value, IndexType indexType) {
-    addHeader(headerListener, name, value, indexType == IndexType.NEVER);
+  private void insertHeader(HeaderListener headerListener, String name, String value, IndexType indexType) {
```

Comment:
There was little value in the `byte[]`-based processing pipeline since ours is based on strings (and also HPACK did some ad-hoc conversions in unexpected places).